spring-statemachine-core/src/main/java/org/springframework/statemachine/access/StateMachineAccessor.java

```diff
@@ -33,11 +34,11 @@
 public interface StateMachineAccessor<S, E> {
 
 	/**
-	 * Execute given {@link StateMachineFunction} with all recursive regions.
+	 * Execute given function with all recursive regions.
 	 *
 	 * @param stateMachineAccess the state machine access
 	 */
-	void doWithAllRegions(StateMachineFunction<StateMachineAccess<S, E>> stateMachineAccess);
+	void doWithAllRegions(Consumer<StateMachineAccess<S, E>> stateMachineAccess);
```

Comment:
Take this comment with a bit of a grain of salt as I didn't try it, but...  Do you think changing signature like this actually allow existing old `StateMachineFunction` to work as is? I was thinking that we might be forced to add a new method which takes `Consumer` which then might actually result compile error due to same parameter type signature with `ambiguous method parameter`.

I'd kinda like not break anything in a user space with `3.0.x`, thus the deprecation which we could then drop with `3.1.x`.

Thought by a major version change(2.x to 3.x) we can actually introduce breaking api changes so I'm questioning my initial idea of letting existing user code to work as is if it turns out to be too complicated. Adding new method name just to have a parameter change from `StateMachineFunction` to `Consumer` doesn't sound that good anyway. What do you think?

Reply:
I think we remove the StateMachineFuntion would be better. In my experience, developers tend to ignore the warning about using deprecated methods or classes. Only compiler error can force them to refactor the code. So even though we add a new method into StateMachineFunction in 3.0.x, developers are more likely to keep using the old method until they get the compiler error after updating to 3.1.x.

Although the compiler error `ambiguous method parameter` caused by adding new method can be solved be type casting, it would make the codes no longer clean.